File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
N = int(input())
result = 0
for _ in range(N):
    word = list(str(input()))
    new_word = [word[0]]
    for i in range(1, len(word)):
        if (word[i] != word[i-1]):
            new_word.append(word[i])
    
    if len(set(new_word)) == len(new_word):
        result += 1

print(result)

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.js
let fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

let n = Number(input[0]);
let result = 0;

for (let i = 1; i < n + 1; i++) {
  let word = input[i];
  let mySet = new Set();
  let isGroupWord = true;
  for (let j = 0; j < word.length; j++) {
    if (word[j] != word[j + 1]) {
      if (mySet.has(word[j])) {
        isGroupWord = false;
        break;
      } else {
        mySet.add(word[j]);
      }
    }
  }

  if (isGroupWord) {
    result += 1;
  }
}

console.log(result);
